content/blog/2018/06/2018-06-26-lessons-java10-hackathon.adoc

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: "What I learnt at the Jenkins & Java 10+ Hackathon"
+tags:
+- events
+- community
+- developer
+- java10
+- java11
+author: tracymiranda
+---
+
+Last week I participated in the link:/blog/2018/06/08/jenkins-java10-hackathon/[Jenkins & Java 10 Online Hackathon].
+It was my first Jenkins hackathon and I roped in link:https://github.com/jonahgraham[Jonah Graham] to do some pair-programming.
+The hackathon, organized by link:https://github.com/oleg-nenashev[Oleg Nenasheva], featured JDK Project Jigsaw committers Mandy Chung and Paul Sandoz, as well as Jenkins creator Kohsuke Kawaguchi. 
+It was a great opportunity for me to learn alot about Jenkins and Java 10.
+
+=== Why Java 10?
+With the link:http://www.oracle.com/technetwork/java/eol-135779.html[Java 8 EoL data looming], the focus was on the current available version of Java, Java 10. 
+Java 10 offers some nice new features and APIs, not least link:https://blog.docker.com/2018/04/improved-docker-container-integration-with-java-10/[improved docker container integration].
+We learnt from Paul of a number of projects with Java 10 migration success stories including Elasticsearch, Kafka & Netty.
+
+At the beginning of the hackathon week, the Jenkins Pipeline feature would crash out when using Java 10. 
+This was resolved with a number of fixes including the upgrade of the link:https://asm.ow2.io/[ASM library].
```

Comment:
also Groovy and JBoss Remoting

Reply:
OK, didn't know this so will leave as is (but good to know)